### DISCUSSION TITLE
Execute custom typescript file with ts-node and sveltekit imports

### DISCUSSION THREAD

**Discussion by kelbyfaessler** [sveltejs/kit#9807] [Q&A] [answered]
I'm using prisma and I have a seed.ts file that loads data into my database for testing. In that file I'd like to do things like

```
import { SOME_VALUE } from "$lib/constants";
```

Notice I'm importing using the $lib alias. Unfortunately when I urn the script using ts-node this import is not found. It also doesn't work if I use a relative import without the alias e.g. `import { SOME_VALUE } from "../src/lib/constants";`.

What do I need to do so my imports will work in my custom scripts?

I added a seed command (`npm run seed`) to my package.json and it looks like this:
```
  "scripts": {
    "dev": "vite dev",
    ...
    "seed": "ts-node --transpile-only src/lib/util/seed.ts",
```

**Comment by rikurainio** [comment]
I have the exact same probem

**Comment by david-plugge** [comment, marked as answer]
Use [vite-node](https://www.npmjs.com/package/vite-node):

```json
{
	"scripts": {
		"seed": "vite-node --options.transformMode.ssr='/.*/' src/lib/util/seed.ts"
	},
	"devDependencies": {
		"vite-node": "^0.30.1"
	}
}
```

`--options.transformMode.ssr='/.*/'` makes sure that vite runs this code as a server, rather than a client. You need this to be able to access private env variables.

**Reply by kelbyfaessler** [reply]
That worked! This is exactly what I thought I needed too as I was looking into the different vite commands but didn't think about other libs. Thanks so much for the help!

**Comment by chokchai** [comment]
this solution works but still errors when using the .env
`import {env} from "$env/dynamic/private"`

**Reply by enisbu** [reply]
Linked to the solution above us, I found that in the seed.ts file -> env variables can be used as `process.env.VARIABLE`

EDIT: found out that the answer above is not needed.
